### PR TITLE
Add crash monitoring handler

### PR DESCRIPTION
**Pull request by e7mac**
The iOS SDK needs to report crashes. We will add another repo which will provide different crash reporting services to the main repo. This commit adds the interface for a general crash handler and looks specifically for a raygun provider

### CODE REVIEW

File: Leanplum-SDK/Classes/LPCrashHandler.m [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if (!sharedCrashHandler)
+            sharedCrashHandler = [[self alloc] init];
+    }
+    return sharedCrashHandler;
+}
+
+-(instancetype)init
+{
+    self = [super init];
+    if (self) {
```
[the remark this answers is not in the document]
Comment (e7mac) [reply Leanplum-SDK/Classes/LPCrashHandler.m]: it's a cocoa thing.
https://www.mikeash.com/pyblog/the-how-and-why-of-cocoa-initializers.html

File: Leanplum-SDK/Classes/LPCrashHandler.m [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        [self initializeRaygunReporter];
+    }
+    return self;
+}
+
+-(void)initializeRaygunReporter
+{
+    _crashReporter = [[NSClassFromString(@"LPRaygunCrashReporter") alloc] init];
```
[the remark this answers is not in the document]
Comment (e7mac) [reply Leanplum-SDK/Classes/LPCrashHandler.m]: No, all properties is backed by _ prefixed instance variables. And in init methods, you can't access properties, but only the backing ivar

File: Leanplum-SDK/Classes/LPCrashHandler.m [hunk 3]
```diff
@@ -0,0 +1,50 @@
+//
+//  LPCrashHandler.m
+//  Leanplum-iOS-SDK-source
+//
+//  Created by Mayank Sanganeria on 6/28/18.
+//
+
+#import "LPCrashHandler.h"
```
[the remark this answers is not in the document]
Comment (e7mac) [reply Leanplum-SDK/Classes/LPCrashHandler.m]: inbuilt in xcode for obj-c code

File: Leanplum-SDK/Classes/LPCrashHandler.m [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+@end
+
+@implementation LPCrashHandler
+
++(instancetype)sharedCrashHandler
+{
+    static LPCrashHandler *sharedCrashHandler = nil;
+    @synchronized(self) {
+        if (!sharedCrashHandler)
```
[the remark this answers is not in the document]
Comment (e7mac) [reply Leanplum-SDK/Classes/LPCrashHandler.m]: that was a miss - i also wanna always have those in there.

**Comment by sayaansaha** [comment]
test this please

**Comment by sayaansaha** [comment]
test this please

**Comment by sayaansaha** [comment]
test this please

**Pull request merged**